Library/Homebrew/dependable.rb

```diff
@@ -3,20 +3,12 @@
 module Dependable
   RESERVED_TAGS = [:build, :optional, :recommended, :run]
 
-  def build?
-    tags.include? :build
-  end
-
-  def optional?
-    tags.include? :optional
-  end
-
-  def recommended?
-    tags.include? :recommended
-  end
+  RESERVED_TAGS.each do |tag|
+    method_name = "#{tag}?"
 
-  def run?
-    tags.include? :run
+    define_method method_name do
+      tags.include? tag
+    end
   end
```

Comment:
I see. Personally, I add a `def method_name` on top to make it searchable. But it's a non-issue since it doesn't introduced improvements per se. Reverted `define_method`